hpqtypes.cabal

```diff
@@ -211,7 +211,7 @@ test-suite hpqtypes-tests
                      , base >= 4.9 && < 5
                      , HUnit >= 1.2
                      , QuickCheck >= 2.5
-                     , aeson >= 0.6.2.0
+                     , aeson >= 0.6.2.0 && < 2.0 || >= 2.0.3.0
```

Comment:
Why is the version between 2.0 and 2.0.3.0 excluded?

Reply:
I think the reason was to reduce the mess (see below) in the `Test.QuickCheck.Arbitrary.Instances` module required to support `aeson` versions `2.0.0.0`, `2.0.1.0`, and `2.0.2.0`.
They contain the new `Object` constructor, but they do not offer the `Arbitrary Value` instance, introduced in version `2.0.3.0` (https://github.com/haskell/aeson/issues/915).

In this way it would be possible to use `aeson >= 1.0 && < 2.1` for tests:
```patch
diff --git a/test/Test/QuickCheck/Arbitrary/Instances.hs b/test/Test/QuickCheck/Arbitrary/Instances.hs
index 142f29f..2b64418 100644
--- a/test/Test/QuickCheck/Arbitrary/Instances.hs
+++ b/test/Test/QuickCheck/Arbitrary/Instances.hs
@@ -12,13 +12,24 @@ import qualified Data.ByteString as BS
 import qualified Data.Text as T
 import qualified Data.UUID.Types as U
 
+-- when aeson version >= 2.0.0
+#if !MIN_VERSION_aeson(2,0,0)
+import qualified Data.HashMap.Strict as HM
+#endif
+
+-- when aeson version < 2.0.3
 #if !MIN_VERSION_aeson(2,0,3)
 import Data.Aeson
 import Data.Scientific
-import qualified Data.HashMap.Strict as HM
 import qualified Data.Vector as V
 #endif
 
+-- when aeson version >= 2.0.0 and < 2.0.3
+#if MIN_VERSION_aeson(2,0,0) && !MIN_VERSION_aeson(2,0,3)
+import qualified Data.Aeson.KeyMap as KM
+import qualified Data.Aeson.Key as Key
+#endif
+
 import Database.PostgreSQL.PQTypes
 
 newtype String0 = String0 { unString0 :: String }
@@ -64,6 +75,7 @@ instance Arbitrary U.UUID where
 
 ----------------------------------------
 
+-- when aeson version < 2.0.3
 #if !MIN_VERSION_aeson(2,0,3)
 instance Arbitrary Scientific where
   arbitrary = scientific <$> arbitrary <*> ((`mod` 100) <$> arbitrary)
@@ -80,7 +92,13 @@ instance Arbitrary Value where
         | otherwise = oneof $ leafs ++ branches
         where
           branches = [
+-- when aeson version >= 2.0.0 and < 2.0.3 (due to the outer macro)
+#if MIN_VERSION_aeson(2,0,0)
+              Object . KM.fromList <$> shortListOf ((,) <$> (Key.fromString <$> arbitrary) <*> subValue)
+-- when aeson version < 2.0.0
+#else
               Object . HM.fromList <$> shortListOf ((,) <$> arbitrary <*> subValue)
+#endif
             , Array  . V.fromList  <$> shortListOf subValue
             ]
           leafs = [
```

Reply:
@marinelli That was exactly the reasons for this exception of these particular `aeson` versions.
I included a comment in the cabal file for clarification.

I also considered more CPP first but went with the simpler solution of just excluding the versions from the test suite.
I am not settled on this and I could implement @marinelli's solution; I am happy with whatever the maintainers prefer.